opensearch_py_ml/ml_models/semantic_highlighter_model.py

```diff
@@ -0,0 +1,350 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+import os
+from typing import List
+from zipfile import ZipFile
+
+import torch
+from torch import nn
+from transformers import AutoTokenizer, BertModel, BertPreTrainedModel
+
+from opensearch_py_ml.ml_commons.ml_common_utils import (
+    _generate_model_content_hash_value,
+)
+from opensearch_py_ml.ml_models.base_models import BaseUploadModel
+
+DEFAULT_MODEL_ID = "opensearch-project/opensearch-semantic-highlighter-v1"
+
+
+class TraceableBertTaggerForSentenceExtractionWithBackoff(BertPreTrainedModel):
+    """
+    A torch.jit-compatible version of the sentence highlighter model.
+    For inference only. Classification logic includes backoff rule
+    with minimum confidence alpha=0.05.
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.bert = BertModel(config)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.classifier = nn.Linear(config.hidden_size, self.num_labels)
+        self.init_weights()
+
+    def forward(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        token_type_ids=None,
+        sentence_ids=None,
+    ):
+        outputs = self.bert(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+        )
+
+        sequence_output = outputs[0]
+        sequence_output = self.dropout(sequence_output)
+
+        @torch.jit.script_if_tracing
+        def _get_agg_output(ids, sequence_output):
```

Comment:
Can we please add more comments if possible? 

Same comment applies for `semantic_highlighter_autotracing.py`.

Reply:
Added docs in code.